test/framework/src/main/java/org/elasticsearch/repositories/blobstore/AbstractBlobContainerRetriesTestCase.java

```diff
@@ -77,15 +78,16 @@ public void tearDown() throws Exception {
     protected abstract BlobContainer createBlobContainer(@Nullable Integer maxRetries,
         @Nullable TimeValue readTimeout,
         @Nullable Boolean disableChunkedEncoding,
-        @Nullable ByteSizeValue bufferSize);
+        @Nullable ByteSizeValue bufferSize,
+        @Nullable BlobPath path);
```

Comment:
I think we shouldn't add this parameter and then ignore it in all the implementations except S3. Can we just set it randomly within this method? It's exposed by the blob container so callers can grab it from there if they need it.

Reply:
It's quite bothersome to do this, this was my first approach last night but it turned into a massively noisy change since we have so many tests that implicitly rely on the path being empty in the container url.
Though technically it would be a nice change to have to get more randomisation on this stuff. Maybe do that in a follow-up (can be right away) and isolate this one since it's quite bad and it goes into 7.14?

Reply:
sure, sounds good